dangerzone/isolation_provider/__init__.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import subprocess
+from abc import ABC, abstractmethod
+from typing import Callable, Optional
+
+from colorama import Fore, Style
+
+from ..document import Document
+
+log = logging.getLogger(__name__)
+
+
+class IsolationProvider(ABC):
+    """
+    Abstracts an isolation provider
+    """
+
+    @abstractmethod
+    def install(self) -> bool:
+        pass
+
+    def convert(
+        self,
+        document: Document,
+        ocr_lang: Optional[str],
+        stdout_callback: Optional[Callable] = None,
+    ) -> None:
+        document.mark_as_converting()
+        try:
+            success = self._convert(document, ocr_lang, stdout_callback)
+        except Exception:
+            success = False
+            log.exception(
+                f"An exception occurred while converting document '{document.id}'"
+            )
```

Comment:
We also need to do `success = False` at the end of this block, since it's not defined. Other than that, looks good.

Reply:
Sorry, I don't follow. I thin it should be set either in `try` or via `except`. Is there any other way this could go?

Reply:
What tha, I thought that there was no `success = False` line in the except block. My eyes must have fooled me, sorry.